src/protocols/dhcpv6/base.c

```diff
@@ -752,23 +752,21 @@ ssize_t	fr_dhcpv6_decode(TALLOC_CTX *ctx, uint8_t const *packet, size_t packet_l
 /** DHCPV6-specific iterator
  *
  */
-void *fr_dhcpv6_next_encodable(void **prev, void *to_eval, void *uctx)
+void *fr_dhcpv6_next_encodable(fr_dlist_head_t *list, void *to_eval, void *uctx)
 {
-	fr_pair_t	*c, *p;
+	fr_pair_t	*c;
 	fr_dict_t	*dict = talloc_get_type_abort(uctx, fr_dict_t);
 
 	if (!to_eval) return NULL;
 
-	for (p = *prev, c = to_eval; c; p = c, c = c->next) {
+	for (c = to_eval; c; c = fr_dlist_next(list, c)) {
```

Comment:
Hm this should really be operating on a `fr_pair_list_t`

Reply:
This is a `dcursor` iterator function, and since dcursors are able to operate on any type of dlist, we have to pass in the `fr_dlist_head` rather than a `fr_pair_list_t`